File: verify/stress11.c
#include "jump.h"
static int b_run_all(int arglist, int rest, int th);
static int b_run_all(int arglist, int rest, int th){
int n,body,save1,save2,save3,goal,cont,res;
save2 = Jget_sp(th);
save3 = Jget_ac(th);
n = Jlength(arglist);
if(n == 0){
loop0:
save1 = Jget_wp(th);
{body = Jwlist3(Jmakeope(","),Jmakecomp("test41"),Jwlist3(Jmakeope(","),Jmakecomp("test42"),Jwlist3(Jmakeope(","),Jmakecomp("test43"),Jwlist3(Jmakeope(","),Jmakecomp("test44"),Jmakecomp("test45"),th),th),th),th);
if((res=Jexec_all(Jaddtail_body(rest,body,th),Jget_sp(th),th)) == YES)
return(YES);
Jset_ac(save3,th);
Junbind(save2,th);
Jset_wp(save1,th);}
return(NO);}
Jerrorcomp(Jmakeint(ARITY_ERR),Jmakecomp("run_all"),arglist);
return(NO);}
static int b_test41(int arglist, int rest, int th);
static int b_test41(int arglist, int rest, int th){
int varX,n,body,save1,save2,save3,goal,cont,res;
save2 = Jget_sp(th);
save3 = Jget_ac(th);
n = Jlength(arglist);
if(n == 0){
loop0:
varX = Jmakevariant(th);
save1 = Jget_wp(th);
{body = Jwlist3(Jmakeope(","),Jwcons(Jmakecomp("p41"),Jwcons(Jmakeint(2),Jwcons(varX,NIL,th),th),th),Jwlist3(Jmakeope(","),Jwcons(Jmakepred("writeln"),Jwcons(Jmakecomp("test41"),NIL,th),th),Jwcons(Jmakepred("writeln"),Jwcons(varX,NIL,th),th),th),th);
if((res=Jexec_all(Jaddtail_body(rest,body,th),Jget_sp(th),th)) == YES)
return(YES);
Jset_ac(save3,th);
Junbind(save2,th);
Jset_wp(save1,th);}
return(NO);}
Jerrorcomp(Jmakeint(ARITY_ERR),Jmakecomp("test41"),arglist);
return(NO);}
static int b_p41(int arglist, int rest, int th);
static int b_p41(int arglist, int rest, int th){
int arg1,arg2,varM,varN,varX,n,body,save1,save2,save3,goal,cont,res;
save2 = Jget_sp(th);
save3 = Jget_ac(th);
n = Jlength(arglist);
if(n == 2){
arg1 = Jnth(arglist,1);
arg2 = Jnth(arglist,2);
loop2:
varM = Jmakevariant(th);
varN = Jmakevariant(th);
varX = Jmakevariant(th);
save1 = Jget_wp(th);
if(Junify_var(arg1,varN,th) == YES && Junify_var(arg2,varX,th) == YES && 1)
if (Jcall(

File: verify/stress11.pl
:- use_module(utils).

run_all :-
    test41, test42, test43, test44, test45.

test41 :-
    p41(2, X),
    writeln(test41),
    writeln(X).


p41(N, X) :-
    ( N > 0 ->
        (M is N - 1,
        p41(M, X))
    ;
        X = done
    ).

test42 :-
    p42(3, 0, X),
    writeln(test42),
    writeln(X).

p42(N, A, X) :-
    ( N > 0 ->
        (B is A + N,
        M is N - 1,
        p42(M, B, X))
    ;
        X = A
    ).

test43 :-
    ( (A=3, A>0) -> X=A ; X=0 ),
    writeln(test43),
    writeln(X).

test44 :-
    ( (A=0, A>0) -> X=A ; X=no ),
    writeln(test44),
    writeln(X).

test45 :-
    ( (A=2 ; A=3) -> B is A + 10 ; B=0 ),
    writeln(test45),
    writeln(B).